src/ActionInteractor.ts

```diff
@@ -57,13 +62,36 @@ export abstract class ActionInteractor {
 
     public async callSkill(serviceRequest: ActionRequest): Promise<any>  {
         const requestJSON = serviceRequest.toJSON();
+
+        if (this.context) {
+            if (requestJSON.queryResult) {
+                //DialogFlow v2
+                requestJSON.queryResult.outputContexts = this.context;
+            } else {
+                //DialogFlow v1
+                requestJSON.result.contexts = this.context;
+            }
+        }
+
+
         if (this.requestFilters.length) {
             this.requestFilters.forEach((requestFilter: RequestFilter) => {
                 requestFilter(requestJSON);
             });
         }
 
-        return this.invoke(requestJSON);
+        const response = await this.invoke(requestJSON);
+
+
+        if (response.contextOut) {
+            // Dialog Flow v1
+            this.context = response.contextOut;
+        } else if (response.outputContexts) {
+            // Dialog Flow v2
+            this.context = response.outputContexts;
+        }
+
+        return response;
```

Comment:
this.context = response.contextOut || response.outputContexts;

Reply:
Can we add a linter that will do these sorts of checks? I know I tend to prefer a more verbose style, but I would mostly just be happy if we had a consistent standard that was applied.

Reply:
@jkelvie added as a ts-lint rule in this repo, we can expand to the next repo's along other agreements.